plugins/candela/web_client/views/CandelaWidget.js

```diff
@@ -44,61 +45,60 @@ var CandelaWidget = View.extend({
     },
 
     render: function () {
-        if (this.accessLevel >= AccessType.READ && this.item.get('name').toLowerCase().endsWith('.csv')) {
-            this.$el.html(CandelaWidgetTemplate({
-                components: this._components
-            }));
-            this.parametersView.setElement($('.g-item-candela-parameters'));
-            datalib.csv(this.item.downloadUrl(), (error, data) => {
-                if (error) {
-                    let info = {
-                        text: 'An error occurred while attempting to read and ' +
-                              'parse the data file. Details have been logged in the console.',
-                        type: 'danger',
-                        timeout: 5000,
-                        icon: 'attention'
-                    };
-                    events.trigger('g:alert', info);
-                    console.error(error);
-                    return;
-                }
+        let parser = null;
+        let name = this.item.get('name').toLowerCase();
+        if (name.endsWith('.csv')) {
+            parser = datalib.csv;
+        } else if (name.endsWith('.tsv') || name.endsWith('.tab')) {
+            parser = datalib.tsv;
+        } else {
+            this.$('.g-item-candela').remove();
+            return;
+        }
 
-                datalib.read(data, {parse: 'auto'});
-
-                // Vega has issues with empty-string fields and fields with dots, so rename those.
-                let rename = [];
-                for (let key in data.__types__) {
-                    if (data.__types__.hasOwnProperty(key)) {
-                        if (key === '') {
-                            rename.push(['', 'id']);
-                        } else if (key.indexOf('.') >= 0) {
-                            rename.push([key, key.replace(/\./g, '_')]);
-                        }
-                    }
-                }
+        this.$el.html(CandelaWidgetTemplate({
+            components: this._components
+        }));
+        this.parametersView.setElement($('.g-item-candela-parameters'));
+        parser(this.item.downloadUrl(), (error, data) => {
+            if (error) {
+                let info = {
```

Comment:
I see no reason for this to be a separate variable. Our style rules allow it to be at the same indentation level inlined into `events.trigger`.